src/main/java/hudson/tasks/junit/JUnitParser.java

```diff
@@ -39,25 +39,38 @@
 
 /**
  * Parse some JUnit xml files and generate a TestResult containing all the
- * results parsed. 
+ * results parsed.
  */
 @Extension
 public class JUnitParser extends TestResultParser {
 
     private final boolean keepLongStdio;
+    private final boolean allowEmptyResults;
 
     /** TODO TestResultParser.all does not seem to ever be called so why must this be an Extension? */
     @Deprecated
     public JUnitParser() {
-        this(false);
+        this(false, false);
     }
 
     /**
      * @param keepLongStdio if true, retain a suite's complete stdout/stderr even if this is huge and the suite passed
      * @since 1.358
      */
+    @Deprecated
```

Comment:
nit: newly deprecated code should also have a javadoc `@deprecated` tag saying what you should use instead.